moveit_core/python/src/moveit/core/__init__.py

```diff
@@ -0,0 +1,5 @@
+# load symbols from C++ extension lib
+from pymoveit_core import *
+
+# and augment with symbols from python modules (order is important to allow overriding!)
```

Comment:
that's a little spooky. why does order matter?

Reply:
Because the python code might want to override what was previously defined in the C++ extension lib. Much like in the following python code, only the second definition "survives":
```python
def foo():
    print("foo")
def foo():
    print("bar")
```
We used this pattern heavily in the past in MoveIt's python wrappers.

Reply:
oh yikes I don't love that :grimacing: can you point me to an example? I'm sure you had a good reason for doing it, but I can't find any examples in moveit_commander.

Reply:
of course we can keep the order as you suggest, it does no harm. So I'll incorporate this into the PR once we figure out the other restructuring stuff we're talking about

Reply:
> can you point me to an example?

`moveit_commander` doesn't do it exactly like this, but very similar. First the C++ extension lib is loaded:
https://github.com/UM-ARM-Lab/moveit/blob/d18173b598846260d4e067ff604c3f3b29366f57/moveit_commander/src/moveit_commander/planning_scene_interface.py#L40
and then its classes (with exactly the same name) are wrapped by python code:
https://github.com/UM-ARM-Lab/moveit/blob/d18173b598846260d4e067ff604c3f3b29366f57/moveit_commander/src/moveit_commander/planning_scene_interface.py#L57
https://github.com/UM-ARM-Lab/moveit/blob/d18173b598846260d4e067ff604c3f3b29366f57/moveit_commander/src/moveit_commander/planning_scene_interface.py#L65-L66

> oh yikes I don't love that

I admit that _overriding_ loaded symbols usually will not make sense. :wink:

Reply:
Ok yea that makes much more sense!